service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/RequestStatementMapper.java

```diff
@@ -0,0 +1,202 @@
+            .isNested(isNested)
+            .availabilityTime(StatementTimeMappingUtils.prepareAvailabilityTimeForReferralRequest(referralRequest))
+            .description(buildDescription(referralRequest))
+            .build();
+
+        if (!referralRequest.hasReasonCode()) {
+            requestStatementTemplateParameters.setDefaultReasonCode(DEFAULT_REASON_CODE_XML);
+        }
+
+        return TemplateUtils.fillTemplate(REQUEST_STATEMENT_TEMPLATE, requestStatementTemplateParameters);
+    }
+
+    private String buildDescription(ReferralRequest referralRequest) {
+        List<String> descriptionList = retrieveDescription(referralRequest);
+        return descriptionList.stream()
+            .filter(StringUtils::isNotEmpty)
+            .collect(Collectors.joining(StringUtils.SPACE));
+    }
+
+    private List<String> retrieveDescription(ReferralRequest referralRequest) {
+        return List.of(
+            buildIdentifierDescription(referralRequest),
+            buildPriorityDescription(referralRequest),
+            buildServiceRequestedDescription(referralRequest),
+            buildRequesterDescription(referralRequest),
+            buildSpecialtyDescription(referralRequest),
+            buildRecipientDescription(referralRequest),
+            buildReasonCodeDescription(referralRequest),
+            buildNoteDescription(referralRequest),
+            buildTextDescription(referralRequest)
+        );
+    }
+
+    private String buildIdentifierDescription(ReferralRequest referralRequest) {
+        Optional<String> identifier = Optional.empty();
+        if (referralRequest.hasIdentifier()) {
+            identifier = referralRequest.getIdentifier()
+                .stream()
+                .filter(val -> val.getSystem().equals(UBRN_SYSTEM_URL))
+                .map(Identifier::getValue)
+                .findFirst();
+        }
+        return identifier.map(value -> UBRN + value).orElse(StringUtils.EMPTY);
+    }
+
+    private String buildPriorityDescription(ReferralRequest referralRequest) {
+        if (referralRequest.hasPriority()) {
+            return PRIORITY + referralRequest.getPriority().getDisplay();
+        }
+        return StringUtils.EMPTY;
+    }
+
+    private String buildServiceRequestedDescription(ReferralRequest referralRequest) {
+        if (referralRequest.hasServiceRequested()) {
+            return SERVICE_REQUESTED + extractServiceRequested(referralRequest);
+        }
+        return StringUtils.EMPTY;
+    }
+
+    private String buildRequesterDescription(ReferralRequest referralRequest) {
+        if (referralRequest.hasRequester() && referralRequest.getRequester().getAgent().hasReference()) {
+            IIdType reference = referralRequest.getRequester().getAgent().getReferenceElement();
+            if (reference.getResourceType().equals(ResourceType.Device.name())) {
```

Comment:
As below, confirm correct behaviour is requester reference cannot be resolved or is an expected type. Do we throw an error and stop processing or is it ok to continue with a blank as done here.

Reply:
Confirmed that if a resource is not resolved an exception should be thrown. 
And that if a reference is not of expected resource type then an exception should also be thrown.